### PR TITLE
Fix base_vrf_selection_test failure for ARISTA

### PR DESCRIPTION
**Pull request by prakashbadri-arista**
The test was failing for the following reasons:
- parent interface was not routed causing the subinterfaces to not be routed as well.
- DUT egress interface ip address was getting configured before the VRF assignment. So when the interface was moved to VRF-10, the interface ipv4 address got reset.
- Unsupported interface ref leaf was set in the interface OC configuration
- The match source ip in the vrf selection policy rule was incorrectly set and hence the incoming traffic flows expected to pass through VRF-10 were not getting forwarded correctly

After fixing the above issues the test is passing correctly

### CODE REVIEW

File: feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ const (
 	vlan20            = 20
 	ipipProtocol      = 4
 	ipv6ipProtocol    = 41
-	ipv4Address       = "198.18.0.1/32"
+	ipv4Address       = "192.0.2.2/32"
```
Comment (Ankur19) [inline feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:46]: This is in the ateSrc IP address space. Update to `192.0.2.13`?

**Review by Ankur19 — CHANGES_REQUESTED** [review]

File: feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go [hunk 2]
```diff
@@ -123,6 +123,9 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice, p1 *ondatra.Port, p2 *on
 	niConfPath := gnmi.OC().NetworkInstance("VRF-10")
 	niConf := configNetworkInstance("VRF-10", i2, 10)
 	gnmi.Replace(t, dut, niConfPath.Config(), niConf)
+	if deviations.InterfaceConfigVRFBeforeAddress(dut) {
+		gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), configInterfaceDUT(i2, &dutDst, 10, vlan10, dut))
```
Comment (Ankur19) [inline feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:127]: `gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), i2)` works as well.

i2 gets modified in-place. Lets not make it more confusing.

File: feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go [hunk 3]
```diff
@@ -145,6 +148,10 @@ func configInterfaceDUT(i *oc.Interface, me *attrs.Attributes, subIntfIndex uint
 	if deviations.InterfaceEnabled(dut) {
 		i.Enabled = ygot.Bool(true)
 	}
+	if deviations.ParentInterfaceRouted(dut) {
```
Comment (Ankur19) [inline feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:151]: `deviations.RequireRoutedSubinterface0(dut)`

Comment (prakashbadri-arista) [reply feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:151]: fixed

File: feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go (hunk 2 above)
Comment (prakashbadri-arista) [reply feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:127]: fixed

File: feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go (hunk 1 above)
Comment (prakashbadri-arista) [reply feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:46]: ok..  I changed the flow srcIp to point to "198.18.0.1" instead

File: feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go [hunk 4]
```diff
@@ -44,6 +44,7 @@ const (
 	ipipProtocol      = 4
 	ipv6ipProtocol    = 41
 	ipv4Address       = "198.18.0.1/32"
+	flowSrcIp         = "198.18.0.1"
```
Comment (Ankur19) [inline feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:47]: Can we revert this for now? We want IP packets both with and without source as "198.18.0.1". I would add a followup PR to add that logic.

File: feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go [hunk 5]
```diff
@@ -338,7 +348,7 @@ func createFlow(name string, top gosnappi.Config, dst attrs.Attributes, innerIpT
 	e1 := flow.Packet().Add().Ethernet()
 	e1.Src().SetValue(ateSrc.MAC)
 	v4 := flow.Packet().Add().Ipv4()
-	v4.Src().SetValue(ateSrc.IPv4)
+	v4.Src().SetValue(flowSrcIp)
```
Comment (Ankur19) [inline feature/experimental/policy/policy_vrf_selection/otg_tests/base_vrf_selection/base_vrf_selection_test.go:351]: Revert.

**Review by Ankur19 — APPROVED** [review]

**Pull request merged**